src/kixi/datastore/web_server.clj

```diff
@@ -195,27 +190,36 @@
      {:consumes "multipart/form-data"
       :produces "application/json"
       :parameters {:body {:file Map
+                          :file-size-bytes s/Str
                           :name s/Str
                           :schema-id s/Str}}
       :part-consumer (map->PartConsumer {:filestore filestore})
       :response (fn [ctx]
                   (let [params (get-in ctx [:parameters :body])
                         file (:file params)
+                        declared-file-size (:file-size-bytes params)
                         metadata {::ms/id (:id file)
                                   ::ss/id (:schema-id params)
                                   ::ms/type "csv"
                                   ::ms/name (:name params)
                                   ::ms/size-bytes (:size-bytes file)
                                   ::ms/provenance {::ms/source "upload"
-                                                   ::ms/pieces-count (:pieces-count file)}}]
+                                                   ::ms/pieces-count (:count file)}}]
                     (let [error (or (spec/explain-data ::ms/filemetadata metadata)
                                     (when-not (ss/exists schemastore (::ss/id metadata))
                                       {::error :unknown-schema
-                                       ::msg {:schema-id (::ss/id metadata)}}))]
+                                       ::msg {:schema-id (::ss/id metadata)}})
+                                    (when-not (= declared-file-size
+                                                 (str (:size-bytes file)))
+                                      {::error :file-upload-failed
+                                       ::msg (str "Expected " declared-file-size " bytes. Got " (:size-bytes file))}))]
                       (if error
                         (return-error ctx error)
                         (do
-                          (c/submit communications metadata)
+                          (c/send-event! communications :kixi.datastore/file-created "1.0.0" metadata)
```

Comment:
It feels like these events should be sent by the filestore or something, rather than the webserver